sorts/recursive_mergesort_array.py

```diff
@@ -0,0 +1,43 @@
+def merge(arr: list[int]) -> list[int]:
+    """Return a sorted array.
+    >>> merge([10,9,8,7,6,5,4,3,2,1])
+    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+    >>> merge([1,2,3,4,5,6,7,8,9,10])
+    [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+    >>> merge([10,22,1,2,3,9,15,23])
+    [1, 2, 3, 9, 10, 15, 22, 23]
+    """
+    if len(arr) > 1:
+        middle_length = len(arr) // 2
+        left_array = arr[:middle_length]
+        right_array = arr[middle_length:]
+        left_size = len(left_array)
+        right_size = len(right_array)
+        merge(left_array)
+        merge(right_array)
+        i = 0
```

Comment:
Please add comments describing the functions of `i`, `j`, and `k`, or make their names more descriptive.